File: lib/js/MMI_Lib/Notification.js
/**
 * notification class
 * 
 * @param spec
 *            The spec object contains all of the information that the
 *            constructor needs to make an instance
 * @returns a new notification object
 */
interaction.notification = function (spec) {
	console.log("[Notification]{interaction.notification} - Started");
	var messageBase = this.messageBase(spec);
	
	// private attribute
	var method = spec.method;
	var name = spec.name;
	var status = spec.status;
	var statusInfo = spec.statusInfo;
	// getter and setter functions
	messageBase.setName = function(aName) {
		name = aName;
	};
	messageBase.getName = function() {
		return name;
	};
	
	messageBase.setStatus = function(aStatus) {
		status = aStatus;
	};
	messageBase.getStatus = function() {
		return status;
	};
	
	messageBase.setStatusInfo = function(aStatusInfo) {
		statusInfo = aStatusInfo;
	};
	messageBase.getStatusInfo = function() {
		return statusInfo;
	};
	
	// getter and setter functions
	messageBase.setMethod = function(aMethod) {
		method = aMethod;
	};
	messageBase.getMethod = function() {
		return method;
	};
	console.log("[Notification]{interaction.notification} - Ended");
	return messageBase;
}

File: lib/java/mmiLib/Notification.java
package mmiLib;

import utilities.Loggable;

public class Notification extends MessageBase implements Loggable {
	private String name;
	private String status;
	private String statusInfo;
	
	public Notification() {
		super();
		this.name = "";
		this.status = "";
		this.statusInfo = "";
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public String getStatusInfo() {
		return statusInfo;
	}

	public void setStatusInfo(String statusInfo) {
		this.statusInfo = statusInfo;
	}
	
	
}
